cmd/skuba/cluster/check.go

```diff
@@ -0,0 +1,133 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubernetesConfigFlags := genericclioptions.NewConfigFlags(true)
+			kubeAdminFile := skuba.KubeConfigAdminFile()
+			kubernetesConfigFlags.KubeConfig = &kubeAdminFile
+
+			config := lib.Config{
+				K8sVersion:      checkOptions.K8sVersion,
+				ForceDownload:   checkOptions.ForceDownload,
+				APIWalk:         checkOptions.APIWalk,
+				SwaggerDir:      checkOptions.SwaggerDir,
+				ShowDescription: checkOptions.ShowDescription,
+				ConfigFlags:     kubernetesConfigFlags,
+				Input:           inputFile,
+			}
+
+			if config.K8sVersion == "" {
+				client, err := kubernetes.GetAdminClientSet()
+				if err != nil {
+					return err
+				}
+				currentClusterVersion, err := kubeadm.GetCurrentClusterVersion(client)
+				if err != nil {
+					return err
+				}
+				availableVersions := kubernetes.AvailableVersions()
+				upgradePath, err := upgrade.CalculateUpgradePath(currentClusterVersion, availableVersions)
+				if err != nil {
+					return err
+				}
+
+				var nextClusterVersion *version.Version
+				if len(upgradePath) > 0 {
+					nextClusterVersion = upgradePath[0]
+				} else {
+					klog.Warning("Already on the latest version, nothing to check.\nFor a specific version use the `kubernetes-version` flag.")
+					return nil
+				}
+				config.K8sVersion = fmt.Sprintf("v%s", nextClusterVersion.String())
+			}
+
+			klog.Infof("Starting Kubepug with configs: %+v", config)
+			kubepug := lib.NewKubepug(config)
+
+			result, err := kubepug.GetDeprecated()
+			if err != nil {
+				return err
+			}
+
+			klog.Info("Starting deprecated objects printing")
+			formatter := formatter.NewFormatter(format)
+			bytes, err := formatter.Output(*result)
+			if err != nil {
+				return err
+			}
+
+			if filename != "" {
+				err = ioutil.WriteFile(filename, bytes, 0644)
+				if err != nil {
+					return err
+				}
+			} else {
+				fmt.Printf("%s", string(bytes))
+			}
+
+			return nil
+		},
+	}
+	cmd.PersistentFlags().BoolVar(&checkOptions.APIWalk, "api-walk", false, "Whether to walk in the whole API, checking if all objects type still exists in the current swagger.json. May be IO intensive to APIServer.")
+	cmd.PersistentFlags().BoolVar(&checkOptions.ShowDescription, "description", true, "Wether to show the description of the deprecated object. The description may contain the solution for the deprecation.")
+	cmd.PersistentFlags().StringVar(&checkOptions.K8sVersion, "kubernetes-version", "", "Which kubernetes release version (https://github.com/kubernetes/kubernetes/releases) should be used to validate objects.")
+	cmd.PersistentFlags().StringVar(&checkOptions.SwaggerDir, "swagger-dir", "", "Where to keep swagger.json downloaded file. If not provided will use the system temporary directory")
+	cmd.PersistentFlags().BoolVar(&checkOptions.ForceDownload, "force-download", false, "Wether to force the download of a new swagger.json file even if one exists.")
```

Comment:
```suggestion
	cmd.PersistentFlags().BoolVar(&checkOptions.ForceDownload, "force-download", false, "Whether to force the download of a new swagger.json file even if one exists.")
```

Reply:
> @evrardjp Are you referring to that Issue/RFC [SUSE/avant-garde#1274](https://github.com/SUSE/avant-garde/issues/1274) ?
> I'm not asking for "Why" we do it. I trust you all on that.
> 
> For QA to approve it, we need to get access to the test (and results) that you developed for the feature.
> If you do not want to write everything from scratch, you can have a look at [caasp-test-cases](https://github.com/SUSE/caasp-test-cases). @atighineanu and I are here to help you.

This is the RFC https://github.com/SUSE/caasp-rfc/pull/64

Reply:
Nice catch, for all the typos.